File: app/src/main/java/com/example/newsapp/domain/model/Article.kt
package com.example.newsapp.domain.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class Article(
    val title: String,
    val author: String,
    val description: String,
    val url: String,
    val imageUrl: String,
    val publishedAt: String,
    val source: String,
    val category: String
) : Parcelable

File: lib/domain/models/article.dart
class Article {
  const Article({
    required this.title,
    required this.author,
    required this.description,
    required this.imageUrl,
    required this.source,
  });

  final String title;
  final String author;
  final String description;
  final String? imageUrl;
  final String source;
}
